### PR TITLE
Enable to open file in ReadWrite mode

### LINKED ISSUE #119: Impossible to set ReadWrite access to file descriptor in FileStream
Hello,it seems there is a bug that does not allow to  open file descriptor in read&write mode despite we have StreamMode::ReadWrite enum value.The problem in the following line of FileStream::open function `_wfopen_s(&fp, path.wstring().c_str(), mode == StreamMode::Read ? L"rb" : L"wb");`.The problem if we pass ReadWrite enum value to FileStream::open method or to constructor it open the file in wb mode and overwrites the content.It is not the problem for load and store func cause as I understood there is no sence to pass FileStream in ReadWrite value to save func cause it has  to clear the content before writing bytes using his own patterns.But it is problem for user who wants to use FileStream as a wrapper to work with files cause he is not able to open file in rw mode using FileStream.I could create pull request but I can not firgure out how to fix it to not break load and store func.

P.S.As an option we can  write something like this `_ _wfopen_s(&fp, path.wstring().c_str(),
     mode == StreamMode::Read ? L"rb" : mode == StreamMode::Write ? L"wb" : L"r+b");` 
but it will break save func cause file will not be cleared before writing our Program;

### PR DESCRIPTION
**Pull request by OptionalAssistant**
Closes #119

### CODE REVIEW

File: src/zasm/src/core/filestream.cpp [hunk 1]
```diff
@@ -88,9 +88,9 @@ namespace zasm
 
         FILE* fp = nullptr;
 #ifdef _WIN32
-        _wfopen_s(&fp, path.wstring().c_str(), mode == StreamMode::Read ? L"rb" : L"wb");
+        _wfopen_s(&fp, path.wstring().c_str(), mode == StreamMode::Read ? L"rb" : mode == StreamMode::Write ? L"wb" : L"r+b");
```
Comment (ZehMatt) [inline src/zasm/src/core/filestream.cpp:91]: I'm not a fan of this, create a new static function that has `static const char* getInternalMode(StreamMode mode)`

File: src/zasm/src/core/filestream.cpp [hunk 2]
```diff
@@ -88,9 +88,9 @@ namespace zasm
...
 #else
-        fp = fopen(path.string().c_str(), mode == StreamMode::Read ? "rb" : "wb");
+        fp = fopen(path.string().c_str(), mode == StreamMode::Read ? "rb" : mode == StreamMode::Write ? "wb" : "r+b");
```
Comment (ZehMatt) [inline src/zasm/src/core/filestream.cpp:93]: Same here, just have `const char* internalMode = getInternalMode(mode)` declared before the preprocessor guard.

**Review by ZehMatt — CHANGES_REQUESTED** [review]

File: src/zasm/src/core/filestream.cpp (hunk 1 above)
Comment (OptionalAssistant) [reply src/zasm/src/core/filestream.cpp:91]: > I'm not a fan of this, create a new static function that has `static const char* getInternalMode(StreamMode mode)`

Ok I did but it seems that _wfopen_s requires wchar_t with L literal and your function return type works only for fopen .Should I create another function ?

**Review by ZehMatt — APPROVED** [review]

**Comment by ZehMatt** [comment]
Thanks again.

**Pull request merged**